src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.CoreCLR.cs

```diff
@@ -668,34 +666,33 @@ public static object CreateWrapperOfType(object o, Type t)
         public static extern int /* ULONG */ Release(IntPtr /* IUnknown */ pUnk);
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern void GetNativeVariantForObject(object obj, /* VARIANT * */ IntPtr pDstNativeVariant);
+        public static extern void GetNativeVariantForObject(object? obj, /* VARIANT * */ IntPtr pDstNativeVariant);
 
         public static void GetNativeVariantForObject<T>(T obj, IntPtr pDstNativeVariant)
         {
-            GetNativeVariantForObject((object)obj, pDstNativeVariant);
+            // TODO-NULLABLE-GENERIC: T can be null
+            GetNativeVariantForObject((object)obj!, pDstNativeVariant);
         }
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern object GetObjectForNativeVariant(/* VARIANT * */ IntPtr pSrcNativeVariant);
+        public static extern object? GetObjectForNativeVariant(/* VARIANT * */ IntPtr pSrcNativeVariant);
```

Comment:
This can return `null` since `MarshalNative::GetObjectForNativeVariant` forwards to `OleVariant::MarshalObjectForOleVariant` which will return null for `VT_EMPTY`